java/impl/gcp/src/main/java/co/worklytics/psoxy/GcpModule.java

```diff
@@ -2,22 +2,73 @@
 
 
 import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.impl.CompositeConfigService;
 import co.worklytics.psoxy.gateway.impl.EnvVarsConfigService;
-import dagger.Binds;
+import com.google.cloud.ServiceOptions;
 import dagger.Module;
+import dagger.Provides;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.time.Duration;
 
 /**
  * defines how to fulfill dependencies that need platform-specific implementations for GCP platform
  */
 @Module
 public interface GcpModule {
 
+    // https://cloud.google.com/functions/docs/configuring/env-var#newer_runtimes
+    enum RuntimeEnvironmentVariables {
+        K_SERVICE
+    }
+
+
+    static String asParameterStoreNamespace(String functionName) {
+        return functionName.toUpperCase().replace("-", "_");
+    }
+
+    // global parameters
+    // singleton to be reused in cloud function container
+    @Provides
+    @Named("Global")
+    @Singleton
+    static SecretManagerConfigService secretManagerConfigService() {
+        // Global don't change that often, use longer TTL
+        return new SecretManagerConfigService(null, ServiceOptions.getDefaultProjectId(), Duration.ofMinutes(20));
```

Comment:
`ServiceOptions.getDefaultProjectId()` is a built-in method from GCP cloud functions framework to provide the `id` of the project. We need that to build the secret names, which could be created using the project id (my-project) or the project number (1234)